File: frontend/src/types/followup.ts
export type FollowUpChannel = 
  | 'PHONE_CALL'
  | 'EMAIL'
  | 'VIDEO_CONFERENCE'
  | 'IN_PERSON_MEETING'
  | 'WHATSAPP_SMS'
  | 'LINKEDIN_MESSAGE'
  | 'OTHER';

export type FollowUpStatus = 
  | 'SCHEDULED'
  | 'COMPLETED'
  | 'MISSED'
  | 'RESCHEDULED'
  | 'CANCELLED';

export type FollowUpOutcome = 
  | 'PENDING'
  | 'INTERESTED'
  | 'PROPOSAL_REQUESTED'
  | 'MEETING_BOOKED'
  | 'CALLBACK_REQUESTED'
  | 'NOT_INTERESTED'
  | 'NO_ANSWER'
  | 'DEAL_WON'
  | 'DEAL_LOST';

export type TargetType = 'LEAD' | 'CUSTOMER' | 'CONTACT';

export type FollowUpPriority = 'LOW' | 'MEDIUM' | 'HIGH' | 'URGENT';

export interface FollowUp {
  id: number;
  title: string;
  channel: FollowUpChannel;
  channelDisplayName: string;
  scheduledAt: string;
  completedAt?: string;
  status: FollowUpStatus;
  statusDisplayName: string;
  outcome: FollowUpOutcome;
  outcomeDisplayName: string;
  priority: FollowUpPriority;
  priorityDisplayName: string;
  notes?: string;
  nextFollowUpDate?: string;
  assignedToUserId?: number;
  assignedToUserName?: string;
  targetType: TargetType;
  targetTypeDisplayName: string;
  targetId?: number;
  targetName?: string;
  productId?: number;
  productName?: string;
  createdByUserId?: number;
  createdByUserName?: string;
  createdByRole?: string;
  isDeleted?: boolean;
  isOverdue: boolean;
  createdAt: string;
  updatedAt?: string;
}

export interface CreateFollowUpRequest {
  title: string;
  channel?: FollowUpChannel;
  scheduledAt: string;
  priority?: FollowUpPriority;
  notes?: string;
  assignedToUserId?: number;
  assignedToUserName?: string;
  targetType?: TargetType;
  targetId?: number;
  targetName?: string;
  productId?: number;
  productName?: string;
}

export interface UpdateFollowUpRequest {
  title: string;
  channel?: FollowUpChannel;
  scheduledAt: string;
  status?: FollowUpStatus;
  outcome?: FollowUpOutcome;
  priority?: FollowUpPriority;
  notes?: string;
  nextFollowUpDate?: string;
  assignedToUserId?: number;
  assignedToUserName?: string;
  targetType?: TargetType;
  targetId?: number;
  targetName?: string;
  productId?: number;
  productName?: string;
}

export interface CompleteFollowUpRequest {
  outcome: FollowUpOutcome;
  notes?: string;
  nextFollowUpDate?: string;
}

export interface RescheduleFollowUpRequest {
  scheduledAt: string;
  notes?: string;
}

export interface FollowUpStats {
  totalFollowUps: number;
  scheduledToday: number;
  totalScheduled: number;
  totalCompleted: number;
  totalMissed: number;
  positiveOutcomes: number;
  successRate: number;
  followUpsByChannel: Record<string, number>;
  followUpsByStatus: Record<string, number>;
  followUpsByOutcome: Record<string, number>;
}

export interface FollowUpCadenceConfig {
  id?: number;
  cadenceName: string;
  initialTouchpointHours: number;
  secondTouchpointDays: number;
  thirdTouchpointDays: number;
  maxAttemptsBeforeDormant: number;
  autoEscalateOverdueHours: number;
  enableSmsReminders: boolean;
  enableEmailCadence: boolean;
  updatedAt?: string;
}

export interface UpdateCadenceConfigRequest {
  cadenceName?: string;
  initialTouchpointHours?: number;
  secondTouchpointDays?: number;
  thirdTouchpointDays?: number;
  maxAttemptsBeforeDormant?: number;
  autoEscalateOverdueHours?: number;
  enableSmsReminders?: boolean;
  enableEmailCadence?: boolean;
}

File: backend/followup-service/src/main/java/com/crm/followup/model/FollowUp.java
package com.crm.followup.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "crm_followups")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class FollowUp {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 200)
    private String title;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 50)
    @Builder.Default
    private FollowUpChannel channel = FollowUpChannel.PHONE_CALL;

    @Column(name = "scheduled_at", nullable = false)
    private LocalDateTime scheduledAt;

    @Column(name = "completed_at")
    private LocalDateTime completedAt;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 50)
    @Builder.Default
    private FollowUpStatus status = FollowUpStatus.SCHEDULED;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 50)
    @Builder.Default
    private FollowUpOutcome outcome = FollowUpOutcome.PENDING;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 50)
    @Builder.Default
    private FollowUpPriority priority = FollowUpPriority.MEDIUM;

    @Column(columnDefinition = "TEXT")
    private String notes;

    @Column(name = "next_follow_up_date")
    private LocalDateTime nextFollowUpDate;

    @Column(name = "assigned_to_user_id")
    private Long assignedToUserId;

    @Column(name = "assigned_to_user_name", length = 120)
    private String assignedToUserName;

    @Enumerated(EnumType.STRING)
    @Column(name = "target_type", nullable = false, length = 50)
    @Builder.Default
    private TargetType targetType = TargetType.LEAD;

    @Column(name = "target_id")
    private Long targetId;

    @Column(name = "target_name", length = 150)
    private String targetName;

    @Column(name = "product_id")
    private Long productId;

    @Column(name = "product_name", length = 200)
    private String productName;

    @Column(name = "created_by_user_id")
    private Long createdByUserId;

    @Column(name = "created_by_user_name", length = 150)
    private String createdByUserName;

    @Column(name = "created_by_role", length = 50)
    private String createdByRole;

    @Column(name = "is_deleted", nullable = false)
    @Builder.Default
    private Boolean isDeleted = false;

    @Column(name = "deleted_at")
    private LocalDateTime deletedAt;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}
